src/hooks/prisma.test.js

```diff
@@ -0,0 +1,191 @@
+import { HandlerInputsBuilder } from '../../testUtils/handlerInputsBuilder';
+import { PrismaSpanBuilder } from '../../testUtils/prismaSpanBuilder';
+import { SpansContainer, TracerGlobals } from '../globals';
+import { getRandomId } from '../utils';
+import { hookPrisma } from './prisma';
+
+describe('Prisma', () => {
+  describe('bad Prisma versions', () => {
+    afterEach(() => {
+      // Reset overrides made to the prisma-client after require() in each test
+      jest.resetModules();
+    });
+
+    let prismaClientLibrary;
+
+    beforeEach(() => {
+      prismaClientLibrary = require('@prisma/client');
+    });
+
+    test('does not fail on no $extends hook', async () => {
+      // Override constructor
+      prismaClientLibrary.PrismaClient = function () {
+        return {
+          $extends: 'not-a-function',
+        };
+      };
+
+      hookPrisma(prismaClientLibrary);
+
+      expect(() => new prismaClientLibrary.PrismaClient()).not.toThrow();
+    });
+
+    test('does not fail when no PrismaClient is missing', async () => {
+      delete prismaClientLibrary.PrismaClient;
+
+      expect(() => hookPrisma(prismaClientLibrary)).not.toThrow();
+    });
+  });
+
+  describe('supported Prisma versions', () => {
+    let client;
+
+    beforeAll(() => {
+      const prismaClientLibrary = require('@prisma/client');
+      hookPrisma(prismaClientLibrary);
+      client = new prismaClientLibrary.PrismaClient();
+    });
+
+    beforeEach(() => {
+      const handlerInputs = new HandlerInputsBuilder().build();
+      TracerGlobals.setHandlerInputs(handlerInputs);
+
+      SpansContainer.clearSpans();
+    });
+
+    afterAll(() => client && client.$disconnect());
+
+    test('produces simple operation span', async () => {
+      const count = await client.user.count({ where: { id: 123456 } });
+
+      const spans = SpansContainer.getSpans();
+      expect(spans).toHaveLength(1);
+
+      const expectedSpan = new PrismaSpanBuilder()
+        .withId(spans[0].id)
+        .withStarted(spans[0].started)
```

Comment:
can you explicitly check the this is actually filled? I.e. that `spans[0].started` is not undefined?
same for other fields

Reply:
also ended 🙏